apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -535,6 +535,11 @@ type ExternalMetricsServerFeatureConfig struct {
 	// +optional
 	Enabled *bool `json:"enabled,omitempty"`
 
+	// RegisterAPIService registers the External Metrics endpoint as an APIService
+	// Default: true
```

Comment:
Could you please add default value to `datadogagent_default.go` and set it in the same file, see this [example](https://github.com/DataDog/datadog-operator/blob/00979d680097751560558961a251831f27fcbf68/apis/datadoghq/v2alpha1/datadogagent_default.go#L75-L77).